src/tests/AddRepoForm.test.js

```diff
@@ -21,3 +21,46 @@ test("should render a link to track first repository if there are no repositorie
   const {container} = render(<AddRepoForm />);
   expect(container).toHaveTextContent("add");
 });
+describe("Test: isValidRepoUrl()", () => {
+  const repoName = "repo_name";
+  const ownerName = "owner_name";
+  const baseUrl = "http://github.com/";
+  const baseSecueUrl = "https://github.com/";
+  test(`repo-url '${baseUrl}${ownerName}/${repoName}' should pass`, () => {
+    const repoUrl = `${baseUrl}${ownerName}/${repoName}`;
+    const [isValid, relativeRepoUrl] = isValidRepoUrl(repoUrl);
+    expect(isValid).toBe(true);
+    expect(relativeRepoUrl).toBe(`${ownerName}/${repoName}`.toLowerCase());
+  });
+  test(`repo-url '${baseSecueUrl}${ownerName}/${repoName}' should pass`, () => {
+    const repoUrl = `${baseSecueUrl}${ownerName}/${repoName}`;
+    const [isValid, relativeRepoUrl] = isValidRepoUrl(repoUrl);
+    expect(isValid).toBe(true);
+
+    expect(relativeRepoUrl).toBe(`${ownerName}/${repoName}`.toLowerCase());
+  });
+  test(`repo-url '${baseUrl}${ownerName}' should fail`, () => {
+    const repoUrl = `${baseUrl}${ownerName}`;
+    const [isValid] = isValidRepoUrl(repoUrl);
+    expect(isValid).toBe(false);
+  });
+  test(`repo-url '${ownerName}/${repoName}' should paas`, () => {
+    const repoUrl = `${ownerName}/${repoName}`;
+    const [isValid, relativeRepoUrl] = isValidRepoUrl(repoUrl);
+    expect(isValid).toBe(true);
+    expect(relativeRepoUrl).toBe(`${ownerName}/${repoName}`);
+  });
+  test(`repo-url '${repoName}' should fail`, () => {
+    const [isValid] = isValidRepoUrl(`${repoName}`);
+    expect(isValid).toBe(false);
+  });
+  test(`repo-url '${ownerName}' should fail`, () => {
+    const [isValid] = isValidRepoUrl(`${ownerName}`);
+    expect(isValid).toBe(false);
+  });
+  test("Case insensitive test", () => {
+    expect(isValidRepoUrl("Http://github.com/owner_name/repo_name")[0]).toBe(true);
+    expect(isValidRepoUrl("Https://github.com/owner_name/repo_name")[0]).toBe(true);
+    expect(isValidRepoUrl("http://GITHUB.com/owner_name/repo_name")[0]).toBe(true);
+  });
+});
```

Comment:
Per my suggestion above, the test is not component-specific and can be moved to a separate file as well.

Reply:
Done. moved all test cases to ```tests/util.test.js```